src/BigQuery/BigQueryClient.php

```diff
@@ -205,71 +181,58 @@ public function __construct(array $config = [])
      *           of results. Setting this flag to a small value such as 1000 and
      *           then paging through results might improve reliability when the
      *           query result set is large.
-     *     @type array $defaultDataset Specifies the default datasetId and
-     *           projectId to assume for any unqualified table names in the
-     *           query. If not set, all table names in the query string must be
-     *           qualified in the format 'datasetId.tableId'.
+     *     @type int $startIndex Zero-based index of the starting row.
      *     @type int $timeoutMs How long to wait for the query to complete, in
      *           milliseconds. **Defaults to** `10000` milliseconds (10 seconds).
-     *     @type bool $useQueryCache Whether to look for the result in the query
-     *           cache.
-     *     @type bool $useLegacySql Specifies whether to use BigQuery's legacy
-     *           SQL dialect for this query. **Defaults to** `true`. If set to
-     *           false, the query will use
-     *           [BigQuery's standard SQL](https://cloud.google.com/bigquery/sql-reference).
+     *     @type int $maxRetries The number of times to retry, checking if the
+     *           query has completed. **Defaults to** `100`.
      *     @type array $parameters Only available for standard SQL queries.
      *           When providing a non-associative array positional parameters
      *           (`?`) will be used. When providing an associative array
      *           named parameters will be used (`@name`).
+     *     @type array $jobConfig Configuration settings for a query job are
+     *           outlined in the [API Docs for `configuration.query`](https://goo.gl/PuRa3I).
+     *           If not provided default settings will be used, with the exception
+     *           of `configuration.query.useLegacySql`, which defaults to `false`
+     *           in this client.
      * }
      * @return QueryResults
+     * @throws \RuntimeException If the maximum number of retries while waiting
```

Comment:
We were asked to provide both high and low level interfaces to interface with a query, `runQuery` was intended to be "high level" (simple, pass a query and get your results), whereas `runQueryAsJob` would be "low level" (more control, but more work required by the user).

Your concern is very fair. Do you have recommendations for how we could fulfill the high level interface without something akin to `runQuery`?